### DISCUSSION TITLE
How to call directly api function to build a readable component

### DISCUSSION THREAD

**Discussion by eguvenc** [dm4t2/vue-currency-input#333] [Q&A]
Hi i want to call directly api function  to create readable component of my CurrencyField.vue 
How ever i get an empty string.
Help me please .

Thanks.

```
<template>
  <span>{{ getText }}</span>
</template>

<script>
import Field from "../../../mixins/field";
import { parse } from 'vue-currency-input'

/**
 * Show value as simple text, render a simple span. HTML tags will be stripped.
 */
export default {
  mixins: [Field],
  props: {
    options: Object,
  },
  computed: {
    getText() {
      return parse(this.value, { currency: 'EUR', precision: 2 })
    },
  },
};
</script>
```

**Comment by dm4t2** [comment]
What means "readable component" and what's its purpose? 
Can you share a minimal reproduction repo or a CodeSandbox?

**Reply by eguvenc** [reply]
Hi i added it to Sandbox. Tested in Microsoft Edge.

My expected value is € 1200,00 but i get just 1200.

https://codesandbox.io/embed/kind-kowalevski-ki2mfb?fontsize=14&hidenavigation=1&theme=dark

![image](https://user-images.githubusercontent.com/482833/175765522-296a91d2-5115-4b59-8d46-164810a91d2b.png)

If you share your mail adress i can add edit permission to CodeSandbox.

eguvenc@gmail.com

**Reply by dm4t2** [reply]
[`parse`](https://dm4t2.github.io/vue-currency-input/api.html#parse) converts a formatted string to a number, but you want to do the opposite.
You can use [Intl.NumberFormat.prototype.format()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/format) for printing currency formatted numbers, see #163

**Comment by eguvenc** [comment]
Thanks. 🤗

On Sat, Jun 25, 2022 at 12:24 PM Matthias Stiller ***@***.***>
wrote:

> parse <https://dm4t2.github.io/vue-currency-input/api.html#parse>
> converts a formatted string to a number, but you want to do the opposite.
> You can use Intl.NumberFormat.prototype.format()
> <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/format>
> for printing currency formatted numbers, see #163
> <https://github.com/dm4t2/vue-currency-input/issues/163>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dm4t2/vue-currency-input/discussions/333#discussioncomment-3021910>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADV4EMA6H4EJIPRJ5V6QW3VQ3F5FANCNFSM5ZYIWUNQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Ersin Güvenç
